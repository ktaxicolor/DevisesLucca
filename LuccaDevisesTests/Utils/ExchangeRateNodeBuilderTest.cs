using LuccaDevises;
using NUnit.Framework;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace LuccaDevisesTests.Utils
{
    [TestFixture]
    public class ExchangeRateNodeBuilderTest
    {
        private FileData data;

        [OneTimeSetUp]
        public void OneTimeSetup()
        {
            data = TestHelper.GetFileData();    
        }

        [Test]
        public void Should_return_root_node_of_built_node_tree()
        {
            var rootOfBuiltTree = ExchangeRateNodeBuilder.BuildFromFileData(data);

            Assert.IsNull(rootOfBuiltTree.Root);
            Assert.IsNotEmpty(rootOfBuiltTree.Children);
        }


        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Should_create_node_tree_given_data_file()
        {
            Approvals.Verify(TestHelper.PrintExchangeNodeTree(ExchangeRateNodeBuilder.BuildFromFileData(data)));
        }
        
}
}
