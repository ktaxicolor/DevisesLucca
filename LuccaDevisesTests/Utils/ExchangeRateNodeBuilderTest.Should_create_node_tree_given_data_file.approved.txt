parent : NULL; initCurrency : ---; finalCurrency : TOP; rate : 0; children count : 1; 
parent : ----TOP; initCurrency : TOP; finalCurrency : VA1; rate : 1,1111; children count : 2; 
parent : TOP-VA1; initCurrency : VA1; finalCurrency : END; rate : 3,3333; children count : 1; 
parent : VA1-END; initCurrency : END; finalCurrency : VA2; rate : 2,2222; children count : 0; 
parent : TOP-VA1; initCurrency : VA1; finalCurrency : VA2; rate : 1,1111; children count : 0; 
